src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/DictionarySlim.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Experimental.Collections
+{
+    public class DictionarySlim<TKey, TValue> : IReadOnlyCollection<KeyValuePair<TKey, TValue>> where TKey : IEquatable<TKey>
+    {
+        const int DefaultPrimeSize = 11;
```

Comment:
How did you pick 11? (Vs regular Dictionary's 3). If 11 is best for k-Nucleotide, it might be better to default to 3 [(or 0, comment)](https://github.com/dotnet/corefxlab/pull/2525/files#r230571231) and have k-Nucleotide pass in 11 if it wants.

Reply:
Correction, Dictionary starts off with no array, then jumps to 3. Since I assume its fairly common to create a dictionary with no specified capacity and not put anything into it, I think it would be best to default to 0/null initially.

Reply:
Have set to 3 for now. I tried 0 but it gives division by zero in the tests. 0/null would both need an additional check for each call. I'll keep on looking at this.